### ISSUE TITLE
[BUG]Groupby agg with np.nan in aggregating column leads to incorrect results 

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#6810]
**Describe the bug**

If we have nulls in the column to run aggregate on we seem to get incorrect results if there are `np.nan` in the column we are aggregating on.    

**Steps/Code to reproduce bug**
```python

>> import numpy as np
>>> import cudf
>>> df = cudf.DataFrame({'key':[0,0,0],'val':[None,1.1,None]})
>>> df['val']=df.val.fillna(np.nan)
>>> df.groupby(by='key').val.sum()
key
0   NaN
Name: val, dtype: float64
```

**Expected behavior**

```python
>>> df.to_pandas().groupby(by='key').val.sum()
key
0    1.1
Name: val, dtype: float64
```

**Environment details**
```python
cudf                      0.17.0a201119   cuda_10.2_py37_g1a80df96c4_285    rapidsai-nightly
libcudf                   0.17.0a201119   cuda10.2_g1a80df96c4_285    rapidsai-nightly
```

**Additional context**
As a work around below does the trick. 

```python
df.nans_to_nulls().groupby(by='key').agg({'val':['sum']})
```

**Comment by kkraus14** [comment]
cc @jrhemstad I think this would require an option in libcudf to control `NaN` behavior in groupby aggregations, but I'm not sure what our thoughts are from the libcudf side.

**Comment by jrhemstad** [comment]
`NaN + x == NaN`

**Comment by vyasr** [comment]
@shwina @brandon-b-miller what do we want to do with something like this? I think with a lot of our more recent discussions around NaN and NULL we have been OK accepting this kind of discrepancy based on the fact that we don't conflate the two, right?

**Comment by shwina** [comment]
I don't think this is related to the NaN v/s N/A discussion. Pandas seems to ignore both in groupby reductions:

```python
In [86]: df = pd.DataFrame({'key':[0,0,0],'val':[pd.NA, 1, pd.NA]})

In [87]: df
Out[87]: 
   key   val
0    0  <NA>
1    0     1
2    0  <NA>

In [88]: df.groupby('key').sum()
Out[88]: 
     val
key     
0      1

In [89]: df = pd.DataFrame({'key':[0,0,0],'val':[np.nan, 1, np.nan]})

In [90]: df
Out[90]: 
   key  val
0    0  NaN
1    0  1.0
2    0  NaN

In [91]: df.groupby('key').sum()
Out[91]: 
     val
key     
0    1.0
```

FWIW, cuDF _also_ ignore nulls and NaNs for plan (i.e., not grouped) reductions:

```python
In [92]: cudf.Series([1, 2, None]).sum()
Out[92]: 3

In [93]: cudf.Series([1, 2, np.nan], nan_as_null=False).sum()
Out[93]: 3.0
```

Note that a `skipna=` kwarg can be used to control null handling in plain reductions:

```python
cudf.Series([1, 2, None]).sum(skipna=False)
nan
```

`GroupBy.sum()` (both in Pandas and cuDF) lacks a similar `skipna=` kwarg. In Pandas, you can work around this using a callable aggregation:

```python
df.groupby('key').agg(lambda x: x.sum(skipna=False))
```

---

Here are a few proposals for how to proceed:

1. We consider this a bug and change our groupby reduction behaviour to ignore nulls, exactly like Pandas
2. We change nothing, and document the difference in behaviour in our [user guide](https://docs.rapids.ai/api/cudf/stable/user_guide/groupby.html).
3. We introduce a `skipna=` argument to group reductions, allowing the user to control whether or not nulls should be skipped, defaulting to `True` to match Pandas behaviour, and allowing `False` to support cuDF's current behaviour:

   ```python
   df.groupby('key').sum(skipna=False)
   ```

We could also ask if Pandas would be interested in supporting (3).
